src/coreclr/vm/i386/stublinkerx86.cpp

```diff
@@ -1289,7 +1289,8 @@ VOID StubLinkerCPU::X86EmitPopRegs(unsigned regSet)
 {
     STANDARD_VM_CONTRACT;
 
-    for (X86Reg r = NumX86Regs; r >= kEAX; r = (X86Reg)(r-1))
+    /* Cmp with regs num cause r is UCHAR */
+    for (X86Reg r = NumX86Regs; r <= NumX86Regs; r = (X86Reg)(r-1))
```

Comment:
This change looks incorrect: the loop is a down-counting loop. The `r <= NumX86Regs` only "accidentally" works because wraparound makes r = 255. Also, it seems like it shouldn't start at `NumX86Regs`. Maybe it would be better to write it as:
```
    for (X86Reg r = NumX86Regs - 1; ; r = (X86Reg)(r-1))
    {
        if (regSet & (1U<<r))
            X86EmitPopReg(r);
        if (r == kEAX)
            break;
    }
```

Also, for `StubLinkerCPU::X86EmitPushRegs`, above, it seems like the loop should be:
```
for (X86Reg r = kEAX; r < NumX86Regs; r = (X86Reg)(r+1))
```

(so, `<`, not `<=`)